contracts/cross-chain/ZNSZChainPortal.sol

```diff
@@ -0,0 +1,221 @@
+            label,
+            tokenURI
+        );
+    }
+
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes calldata
+    ) external pure override returns (bytes4) {
+        return this.onERC721Received.selector;
+    }
+
+    function setDestZnsPortal(address newAddress) external override onlyAdmin {
+        if (newAddress == address(0)) revert ZeroAddressPassed();
+
+        destZnsPortal = newAddress;
+
+        emit DestZnsPortalSet(newAddress);
+    }
+
+    function _processPayment(
+        string calldata label,
+        bytes32 parentHash
+    ) internal {
+        // TODO multi: do we actually want to stake on BOTH sides? What other payment option can we do ???!!!
+        // TODO multi: if we leave this option (stake as Portal address) this needs to be optimized!!!
+        // take payment to this contract so it can register
+
+        uint256 price;
+        uint256 stakeFee;
+        IZNSPricer rootPricer = rootRegistrar.rootPricer();
+        if (parentHash == bytes32(0)) { // Root Domains
+            price = rootPricer.getPrice(
+                parentHash,
+                label,
+                true
+            );
+        } else { // Subdomains
+            (IZNSPricer pricer, PaymentType paymentType,) = subRegistrar.distrConfigs(parentHash);
+            (price, stakeFee) = pricer.getPriceAndFee(
+                parentHash,
+                label,
+                true
+            );
+            stakeFee = paymentType == PaymentType.STAKE ? stakeFee : 0;
+        }
+
+        uint256 protocolFee = rootPricer.getFeeForPrice(bytes32(0), price + stakeFee);
+        uint256 totalCost = price + stakeFee + protocolFee;
+
+        ( IERC20 paymentToken, ) = treasury.paymentConfigs(parentHash);
+        paymentToken.transferFrom(msg.sender, address(this), totalCost);
+
+        paymentToken.approve(address(treasury), totalCost);
+    }
+
+    function _bridgeDomain(
+        bytes32 domainHash,
+        bytes32 parentHash,
+        string calldata label,
+        string calldata tokenURI
+    ) internal {
+        // Create data proof for ZNS on L2
+        // we are using msg.sender here because the current caller on L1 will be the owner of the bridged L2 domain
+        BridgedDomain memory proof = BridgedDomain({
+            domainOwner: msg.sender,
+            domainHash: domainHash,
+            parentHash: parentHash,
+            label: label,
+            tokenUri: tokenURI
+        });
+
+        bytes memory encodedProof = abi.encode(proof);
+
+        // TODO multi: should we leave this? We can't rely on setting this properly on time
+        //  and can permanently lock domains !!!
+        if (destZnsPortal == address(0)) revert DestinationPortalNotSetInState();
+
+        polygonZkEVMBridge.bridgeMessage(
+            destNetworkId,
+            destZnsPortal,
+            // TODO multi: figure out what this is and how to better pass it !!!
```

Comment:
is this comment in reference to a similar comment for `forceGlobalExitRoot` above on L74?